src/Symfony/Component/Routing/Tests/RouteCollectionTest.php

```diff
@@ -302,4 +302,17 @@ public function testSetMethods()
         $this->assertEquals(array('PUT'), $routea->getMethods());
         $this->assertEquals(array('PUT'), $routeb->getMethods());
     }
+
+    public function testAddNamePrefix()
+    {
+        $collection = new RouteCollection();
+        $collection->add('foo', $foo = new Route('/foo'));
```

Comment:
please also add an existing `api_foo` route, to be sure that the prefixing does not delete this route due to overriding.

Reply:
Good catch, added.